src/tabular-api-surface.ts

```diff
@@ -736,12 +759,30 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
         err = null;
       }
 
+      const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
       if (mutationErrorsByEntryIndex.size !== 0) {
-        const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
         callback(new PartialFailureError(mutationErrors, err));
         return;
       }
-
+      if (err && timeoutExceeded) {
+        callback(
+          new PartialFailureError(
+            /* We concatenate an error onto the list for each pending entry
+              because it is required in TestMutateRows_Generic_DeadlineExceeded.
+              The test checks to see if the number of entries is greater than 0
+              indicating that it is expecting some errors even though no data
+              has been received yet.
+               */
```

Comment:
It is expecting at least one error reported [here](https://github.com/googleapis/cloud-bigtable-clients-test/blob/c78ca1867bbf862ce7d742a999b7d45ae5edc478/tests/mutaterows_test.go#L227).

Reply:
Yes, partialFalureError should contain a list of failed entries. For example if we sent a requet [row1, row2, row3] and they all failed with unavailable errors, the failure will look something like: DeadlineExceeded(unavailable, unavailable, unavailable)

Reply:
Thanks for the example.

The behaviour of the test is that it makes a request and then [no errors for entries are sent back while one entry is still pending](https://github.com/googleapis/cloud-bigtable-clients-test/blob/c78ca1867bbf862ce7d742a999b7d45ae5edc478/tests/mutaterows_test.go#L197). Then the client receives a `DEADLINE_EXCEEDED` error for the test overall. So without any code changes we get DeadlineExceeded(... no responses yet).

My change takes the pending responses and turns them into DEADLINE_EXCEEDED errors at the entry level because this is what the test is expecting. I want to double check this with you though because it seems like an unnatural fix to get the test to pass.